src/Microsoft.OData.Core/Json/BufferingJsonReader.cs

```diff
@@ -235,6 +249,111 @@ public bool Read()
             return this.ReadInternal();
         }
 
+        /// <summary>
+        /// The value of the last reported node.
+        /// </summary>
+        /// <remarks>
+        /// Depending on whether buffering is on or off this will return the node type of the last
+        /// buffered read or the node type of the last unbuffered read.
+        /// </remarks>
+        // TODO: Document <returns/>
+        public Task<object> GetValueAsync()
+        {
+            if (this.bufferedNodesHead != null)
+            {
+                if (this.isBuffering)
+                {
+                    Debug.Assert(this.currentBufferedNode != null, "this.currentBufferedNode != null");
+                    return Task.FromResult(this.currentBufferedNode.Value);

```

Comment:
Out of curiosity, you opted not to use ValueTask in this case because this is a public method or because the synchronous code-path is less frequent than the async one?